packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/connector/ConnectorContextualMenu.tsx

```diff
@@ -174,13 +174,13 @@ export const ConnectorContextualMenu = ({}: ConnectorContextualMenuProps) => {
       }
     }
     if (invokeSingleClickOnTwoDiagramElementToolError) {
-      addErrorMessage(error.message);
+      addErrorMessage(error?.message ?? 'An unexpected error has occurred, please refresh the page');
     }
   }, [invokeSingleClickOnTwoDiagramElementToolData, invokeSingleClickOnTwoDiagramElementToolError]);
 
   const connectorTools: GQLTool[] = [];
   if ((data?.viewer.editingContext.representation.description.__typename ?? '') === 'DiagramDescription') {
-    (data.viewer.editingContext.representation.description as GQLDiagramDescription).connectorTools.forEach((tool) =>
+    (data!.viewer.editingContext.representation.description as GQLDiagramDescription).connectorTools.forEach((tool) =>
```

Comment:
Improve the test above to ensure that data is considered non nullable here

Reply:
Done